django/utils/html.py

```diff
@@ -17,7 +17,7 @@
 from .html_parser import HTMLParseError, HTMLParser
 
 # Configuration for urlize() function.
-TRAILING_PUNCTUATION = ['.', ',', ':', ';', '.)', '"', '\'', '!']
+TRAILING_PUNCTUATION_RE = re.compile(r'^(.*?)([.,:;!]+)$')
```

Comment:
> It seems like some characters that were in `TRAILING_PUNCTUATION` aren't in the regex -- could you explain that to me so I understand it?

I dropped the closing parenthesis, single quote, and double quote as they are handled by `WRAPPING_PUNCTUATION`. If I missed a detail as to why these need to exist twice, let me know and I'll add a test case for it.